### PR TITLE
Allow users to correct predictions and export a modified CSV

### LINKED ISSUE #135: Allow users to correct classified labels and export a modified CSV
In the observations explorer (opened by clicking images displayed over the map), the user should be able to override the top prediction by selecting one of the available classes or providing a custom one.

The user should then have an option to export a CSV file with modified predictions. The exported CSV should have the following columns modified:
* common_name: custom prediction
* taxon columns: matched from taxons.csv or empty if not available
* uncertainty: 1
* identified_by: user

### PR DESCRIPTION
**Pull request by kamilzyla**
Closes #135 

### Description
Implementing this feature required quite a bit of refactoring. I resolved several issues along the way (missing `key` props, invalid nested components, map reloading on inspector close).

I'm also introducing some technical debt though: the `constants/taxons.ts` file is a copy of data in `taxons.csv`. The file needs to be available both for the Python script and JS frontend and we don't have a good solution for sharing assets between them. It was the simplest, robust way.

### Definition of Done
* [x] The change has a corresponding task. Each task needs to be well defined and described.
* [x] Documentation (README, comments, etc.) is updated accordingly.
* [x] Change has been tested (manually or with automated tests).

### CODE REVIEW

File: app/components/ObservationsInspector.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  };
+
+  const predictions = [
+    [formatAnimalClassName(observation.pred_1), observation.score_1],
+    [formatAnimalClassName(observation.pred_2), observation.score_2],
+    [formatAnimalClassName(observation.pred_3), observation.score_3]
+  ];
+
+  return (
+    <Card
```
Comment (marekrogala) [inline app/components/ObservationsInspector.tsx:38]: I'd suggest splitting this into several variables that you later merge to make the high level structure more readable in the final return

File: app/constants/ObservationsData.ts [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
+  identified_by: string;
+  wi_taxon_id: string;
+  class: string;
+  order: string;
+  family: string;
+  genus: string;
+  species: string;
+  common_name: string;
+  uncertainty: number;
+  // Metedata columns
```
Comment (marekrogala) [inline app/constants/ObservationsData.ts:15]: typo

File: app/constants/ObservationsData.ts [hunk 3]
```diff
@@ -1,19 +1,31 @@
 type Observation = {
+  // WCS columns
```
Comment (marekrogala) [inline app/constants/ObservationsData.ts:2]: minor thing: i understand where "columns" comes from, but it doesn't fully make sense in this place. Maybe "attributes" or something like that

File: app/containers/ExplorePage.tsx [hunk 4]
```diff
@@ -90,19 +156,36 @@ export default function ExplorePage() {
     certaintyRange: [0, 1]
   });
   const [data, setData] = useState<undefined | ObservationsData>();
+  const [inspectedObservations, setInspectedObservations] = useState<
```
Comment (marekrogala) [inline app/containers/ExplorePage.tsx:159]: i suppose this styling is enforced by some rules, but makes me sad to see this broken into 3 lines like this... Doesn't really increase readability. if possible try to fit this in one line? maybe we need to increase line length limit? or use some shorter name for this part of state

File: app/containers/ExplorePage.tsx [hunk 5]
```diff
@@ -90,19 +156,36 @@ export default function ExplorePage() {
...
+    Observation[]
+  >([]);
+  const [predictionOverrides, setPredictionOverrides] = useState<
```
Comment (marekrogala) [inline app/containers/ExplorePage.tsx:162]: ditto

File: app/containers/ExplorePage.tsx [hunk 6]
```diff
@@ -90,19 +156,36 @@ export default function ExplorePage() {
...
   const handleFilters = (val: string[]) => {
     setFilters({ ...filters, ...val });
   };
 
+  const handlePredictionOverride = (
+    location: string,
+    override: CreatableOption
+  ) => {
+    const overrides = { ...predictionOverrides, [location]: override };
+    if (!override) {
```
Comment (marekrogala) [inline app/containers/ExplorePage.tsx:175]: I'm not sure i understand this fully (I don't know about CreatableOption) but looks a bit suspicious and Python-like... In Javascript I'd expect full comparison with `!==`. using just a negation suggests this is a bool but I don't see this in the

File: app/components/ObservationsInspector.tsx (hunk 1 above)
Comment (kamilzyla) [reply app/components/ObservationsInspector.tsx:38]: Done.

File: app/constants/ObservationsData.ts (hunk 2 above)
Comment (kamilzyla) [reply app/constants/ObservationsData.ts:15]: Done.

File: app/constants/ObservationsData.ts (hunk 3 above)
Comment (kamilzyla) [reply app/constants/ObservationsData.ts:2]: Done.

File: app/containers/ExplorePage.tsx (hunk 6 above)
Comment (kamilzyla) [reply app/containers/ExplorePage.tsx:175]: Good point, refactored the code to make it more clear what's happening.

File: app/containers/ExplorePage.tsx (hunk 4 above)
Comment (kamilzyla) [reply app/containers/ExplorePage.tsx:159]: Done; I hated this too :)

File: app/containers/ExplorePage.tsx (hunk 5 above)
Comment (kamilzyla) [reply app/containers/ExplorePage.tsx:162]: Done.

**Pull request merged**